### PR TITLE
create the ealapps role

### PR DESCRIPTION
**Pull request by kayiwa**
adds a deploy user with keys to be used to clone.
configures php and apache2

### CODE REVIEW

File: roles/ealapps/files/eal_app [hunk 1]
```diff
@@ -0,0 +1,26 @@
+$ANSIBLE_VAULT;1.1;AES256
```
Comment (sandbergja) [inline roles/ealapps/files/eal_app:1]: What does this private key do?  And can we add a `.key` to the filename?

File: roles/ealapps/molecule/default/verify.yml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  gather_facts: false
+  tasks:
+  - name: check php curl status
+    ansible.builtin.package:
+      name: "{{ item }}"
+      state: present
+    check_mode: true
+    register: pkg_status
+    loop:
+      - php8.1-curl
```
Comment (sandbergja) [inline roles/ealapps/molecule/default/verify.yml:13]: Why are we just checking for php8.1-curl, and not the other php8.1 packages (like php8.1, php8.1-dev, php8.1-zip) here?

File: roles/ealapps/molecule/default/verify.yml [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+
+  - name: test for ruby packages
```
Comment (sandbergja) [inline roles/ealapps/molecule/default/verify.yml:15]: What does this ruby packages test do?

File: roles/ealapps/tasks/main.yml [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+    state: present
+
+- name: ealapps | install github deploy keys
+  ansible.builtin.template:
+    src: "{{ item }}"
+    dest: "/home/{{ deploy_user }}/.ssh/{{ item }}"
+    owner: "{{ deploy_user }}"
+    group: "{{ deploy_user }}"
+    mode: 0600
+  with_items: "{{ deploy_keys }}"
```
Comment (sandbergja) [inline roles/ealapps/tasks/main.yml:15]: Where is the deploy_keys variable defined?

**Review by sandbergja — CHANGES_REQUESTED** [review]
Thanks for your work on this, @kayiwa !  @christinach and I reviewed the first half and we'll continue tomorrow.

File: roles/ealapps/molecule/default/verify.yml (hunk 2 above)
Comment (kayiwa) [reply roles/ealapps/molecule/default/verify.yml:13]: this come from the PHP role. https://github.com/pulibrary/princeton_ansible/blob/674f6dc4c41fcf91686fa30c03620340d1b1fe9a/roles/php/tasks/main.yml#L19 and I notice curl is already in there.

File: roles/ojs/templates/script.exp.j2 [hunk 5]
```diff
@@ -43,7 +43,7 @@ if {$force_conservative} {
 
 
 set timeout -1
-spawn php7.4 tools/install.php
+spawn php8.1 tools/install.php
```
Comment (christinach) [inline roles/ojs/templates/script.exp.j2:46]: @sandbergja was wondering if ojs should be protected at php7?

File: roles/php/tasks/configure-apache2.yml [hunk 6]
```diff
@@ -16,18 +16,3 @@
...
-  notify: restart apache
-  when:
-    - running_on_server
-  ignore_errors: true
-
-- name: php | enable php apache module
-  apache2_module:
-    state: present
-    name: php{{ php_version }}
-  notify: restart {{ php_webserver }}
```
Comment (christinach) [inline roles/php/tasks/configure-apache2.yml:33]: Why do we remove these tasks?

**Review by christinach — CHANGES_REQUESTED** [review]
@kayiwa, @sandbergja and I added a couple more questions.

File: roles/php/tasks/configure-apache2.yml (hunk 6 above)
Comment (kayiwa) [reply roles/php/tasks/configure-apache2.yml:33]: ansible does the right thing and installs the module.

File: roles/ealapps/files/eal_app (hunk 1 above)
Comment (kayiwa) [reply roles/ealapps/files/eal_app:1]: allows us to clone a private repo.

File: roles/ealapps/tasks/main.yml (hunk 4 above)
Comment (kayiwa) [reply roles/ealapps/tasks/main.yml:15]: https://github.com/pulibrary/princeton_ansible/blob/eaf7664c9cb9d2cd3cac805498f4a203274ae744/roles/deploy_user/defaults/main.yml#L3

File: roles/ojs/templates/script.exp.j2 (hunk 5 above)
Comment (kayiwa) [reply roles/ojs/templates/script.exp.j2:46]: I got the blessing of RDSS to move on with this

Comment (kayiwa) [inline roles/ojs/templates/script.exp.j2:46]: can we take advantage of templates with "{{ php_version }}" for future us

File: playbooks/ealapps.yml [hunk 7]
```diff
@@ -0,0 +1,19 @@
+---
+# by default this playbook runs in the staging environment
```
Comment (kayiwa) [inline playbooks/ealapps.yml:2]: add this to the apache service

File: roles/drupal9/molecule/default/verify.yml [hunk 8]
```diff
@@ -40,16 +40,25 @@
     register: file
     failed_when:
       - not file.stat.exists
+  - name: remove legacy php packages
```
Comment (kayiwa) [inline roles/drupal9/molecule/default/verify.yml:43]: this only allows it to pass on CI. What about the VMS

File: roles/ealapps/README.md [hunk 9]
```diff
@@ -0,0 +1,38 @@
+Role Name
+=========
+
+A brief description of the role goes here.
+
```
Comment (kayiwa) [inline roles/ealapps/README.md:5]: add documentation about role

File: roles/ealapps/files/id_rsa.pub [hunk 10]
```diff
@@ -0,0 +1 @@
+ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABgQC2RPcHrYslQj6AHe+AHai2HRgy8Z0qOo14xDbvO6g7yFJd2pfoG+UgbscgPwl8vRpD9TFuTwkcd9BE9hiBQW6oiVOdbADMAIobXK2AmNKN0vzHLs3UZRqvnq9loKUswxMKGUD/2sfCV+nSVpQYj8ZIwW+XmJinY7T2yttChXlWa3BLqTDt0T+A8vFbBYC03Kre278ypGDmuAkuvDpXI0QW9RXJxuyoxT39TaJmh0R0a66q9y2VC5ElOeheDyw9QP7hl7IYi2M3/GFO5mOjLzZNkcwJ+A29zxSXl0cnmZX6S9bxgohIOBinXhz3/g9DfB2wJfyBlGjiqE7Ut8M02P7E6Q/LaiQLK+St8oDTO0yFA/QsgvPgLXVynfiu1mjXix6to1O719jabwjs2XBFBD/zIoawMu3FAId8Y5Sqjm36LI1JZLdYLD+OxFJwlBauPwiIunrQ5sIEjxNAOfLC4jaj9i8+ol00ldi0lNVMXyZ6SKj+5Wbhm3Lw3A8/YPA5VNE= kayiwa@PU-F5KSY01XF9VN
```
Comment (kayiwa) [inline roles/ealapps/files/id_rsa.pub:1]: we don't need both.

File: roles/ealapps/meta/main.yml [hunk 11]
```diff
@@ -0,0 +1,19 @@
+---
+galaxy_info:
+  role_name: ealapps
+  company: Princeton University Library
+  description: <Role Description>
```
Comment (kayiwa) [inline roles/ealapps/meta/main.yml:5]: add a description

File: roles/ealapps/tasks/main.yml [hunk 12]
```diff
@@ -0,0 +1,66 @@
+---
+# tasks file for ealapps
+- name: ealapps | install ealapps dependencies
+  ansible.builtin.apt:
+    name: ["php{{ php_version }}-curl"]
```
Comment (kayiwa) [inline roles/ealapps/tasks/main.yml:5]: php role installs this already.

File: roles/ealapps/tests/inventory [hunk 13]
```diff
@@ -0,0 +1,2 @@
+localhost
```
Comment (kayiwa) [inline roles/ealapps/tests/inventory:1]: we don't need this from `molecule init role`

File: roles/ojs/templates/script.exp.j2 (hunk 5 above)
Comment (kayiwa) [inline roles/ojs/templates/script.exp.j2:46]: use "{{ php_version }}"

**Review by christinach — CHANGES_REQUESTED** [review]
@kayiwa we need to install `php8.1-gd` and `php8.1-mysql` package in the drupal role. We tested this with @sandbergja and @kevinreiss.

**Review by christinach — CHANGES_REQUESTED** [review]
@kayiwa we would also like to install `php8.1-mbstring` for the drupal role. Thanks!

**Comment by sandbergja** [comment]
@kayiwa @christinach I think these lines can just be updated to reference {{ php_version }}, rather than hard-coded php7.4: https://github.com/pulibrary/princeton_ansible/blob/main/roles/drupal/tasks/main.yml#L7-L10

That will also bring in the `php8.1-xml` package, which is needed for the special_collections site.

File: roles/libwww/molecule/default/verify.yml [hunk 14]
```diff
@@ -1,8 +1,9 @@
 ---
 # This is an example playbook to execute Ansible tests.
-
 - name: Verify
   hosts: all
+  vars:
+    - php_version: "8.1"
```
Comment (kayiwa) [inline roles/libwww/molecule/default/verify.yml:6]: change this back to 7.4

File: playbooks/ealapps.yml [hunk 15]
```diff
@@ -0,0 +1,19 @@
...
+# to run in production, pass '-e runtime_env=production'
+- name: build the ealapps
+  hosts: ealapps_{{ runtime_env | default('staging') }}
+  remote_user: pulsys
+  become: true
+  vars_files:
+    - ../group_vars/ealapps/{{ runtime_env | default('staging') }}.yml
+    - ../group_vars/ealapps/vault.yml
```
Comment (sandbergja) [inline playbooks/ealapps.yml:10]: This file isn't yet in source control, so the playbook fails with:

```
PLAY [build the ealapps] ***********************************************************************************************************************************
ERROR! vars file ../group_vars/ealapps/vault.yml was not found
Could not find file on the Ansible Controller.
If you are using a module and expect the file to exist on the remote, see the remote_src option
```

**Review by sandbergja — APPROVED** [review]
Thanks so much, @kayiwa

**Review by christinach — APPROVED** [review]
Thanks @kayiwa !

**Pull request merged**